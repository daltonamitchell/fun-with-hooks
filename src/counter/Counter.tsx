import React, { Component } from "react";

class Counter extends Component {
  state = { count: 0 };

  setCount = () => {
    this.setState({ count: this.state.count + 1 });
  };

  render() {
    return (
      <div>
        <p>You clicked {this.state.count} times</p>
        <button onClick={this.setCount}>Click me</button>
      </div>
    );
  }
}

export default Counter;
