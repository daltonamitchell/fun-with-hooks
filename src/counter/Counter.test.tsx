import React from "react";
import { shallow, ShallowWrapper } from "enzyme";

import Counter from "./CounterFull";

describe("<Counter/>", () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<Counter />);
  });

  it("renders without crashing", () => {
    expect(component.debug()).toMatchSnapshot();
  });

  it("displays the initial count", () => {
    expect(component.text()).toContain("Current count is 0");
    expect(component.debug()).toMatchSnapshot();
  });

  it("counts up", () => {
    component.find("[data-testid='up']").simulate("click");
    expect(component.text()).toContain("Current count is 1");
    expect(component.debug()).toMatchSnapshot();
  });

  it("counts down", () => {
    component.find("[data-testid='down']").simulate("click");
    expect(component.text()).toContain("Current count is -1");
    expect(component.debug()).toMatchSnapshot();
  });

  it("resets count", () => {
    component.find("[data-testid='up']").simulate("click");
    component.find("[data-testid='up']").simulate("click");
    expect(component.text()).toContain("Current count is 2");
    expect(component.debug()).toMatchSnapshot();

    component.find("[data-testid='reset']").simulate("click");
    expect(component.text()).toContain("Current count is 0");
    expect(component.debug()).toMatchSnapshot();
  });
});
