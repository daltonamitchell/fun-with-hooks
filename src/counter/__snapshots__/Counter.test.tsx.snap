// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> counts down 1`] = `
"<div>
  <p>
    Current count is 
    -1
  </p>
  <button data-testid=\\"up\\" onClick={[Function: onClick]}>
    Count Up
  </button>
  <button data-testid=\\"down\\" onClick={[Function: onClick]}>
    Count Down
  </button>
  <br />
  <button data-testid=\\"reset\\" onClick={[Function: onClick]}>
    Reset Count
  </button>
</div>"
`;

exports[`<Counter/> counts up 1`] = `
"<div>
  <p>
    Current count is 
    1
  </p>
  <button data-testid=\\"up\\" onClick={[Function: onClick]}>
    Count Up
  </button>
  <button data-testid=\\"down\\" onClick={[Function: onClick]}>
    Count Down
  </button>
  <br />
  <button data-testid=\\"reset\\" onClick={[Function: onClick]}>
    Reset Count
  </button>
</div>"
`;

exports[`<Counter/> displays the initial count 1`] = `
"<div>
  <p>
    Current count is 
    0
  </p>
  <button data-testid=\\"up\\" onClick={[Function: onClick]}>
    Count Up
  </button>
  <button data-testid=\\"down\\" onClick={[Function: onClick]}>
    Count Down
  </button>
  <br />
  <button data-testid=\\"reset\\" onClick={[Function: onClick]}>
    Reset Count
  </button>
</div>"
`;

exports[`<Counter/> renders without crashing 1`] = `
"<div>
  <p>
    Current count is 
    0
  </p>
  <button data-testid=\\"up\\" onClick={[Function: onClick]}>
    Count Up
  </button>
  <button data-testid=\\"down\\" onClick={[Function: onClick]}>
    Count Down
  </button>
  <br />
  <button data-testid=\\"reset\\" onClick={[Function: onClick]}>
    Reset Count
  </button>
</div>"
`;

exports[`<Counter/> resets count 1`] = `
"<div>
  <p>
    Current count is 
    2
  </p>
  <button data-testid=\\"up\\" onClick={[Function: onClick]}>
    Count Up
  </button>
  <button data-testid=\\"down\\" onClick={[Function: onClick]}>
    Count Down
  </button>
  <br />
  <button data-testid=\\"reset\\" onClick={[Function: onClick]}>
    Reset Count
  </button>
</div>"
`;

exports[`<Counter/> resets count 2`] = `
"<div>
  <p>
    Current count is 
    0
  </p>
  <button data-testid=\\"up\\" onClick={[Function: onClick]}>
    Count Up
  </button>
  <button data-testid=\\"down\\" onClick={[Function: onClick]}>
    Count Down
  </button>
  <br />
  <button data-testid=\\"reset\\" onClick={[Function: onClick]}>
    Reset Count
  </button>
</div>"
`;
